datafusion/functions-aggregate/src/sum.rs

```diff
@@ -143,11 +166,19 @@ impl AggregateUDFImpl for Sum {
                 dt if dt.is_signed_integer() => Ok(DataType::Int64),
                 dt if dt.is_unsigned_integer() => Ok(DataType::UInt64),
                 dt if dt.is_floating() => Ok(DataType::Float64),
-                _ => exec_err!("Sum not supported for {}", data_type),
+                _ => exec_err!("Sum not supported for {}", data_type).map_err(|err| {
+                    let diagnostic = Diagnostic::new_error(
+                        format!(
+                            "Coercion only supports integer and floating point values"
+                        ),
+                        sum.spans().first(),
+                    );
```

Comment:
I don't think "Coercion only supports integer and floating point values" is a message that an end user of an application built on DataFusion can understand. Perhaps something like `'sum' only works on numeric values`?